src/windowing.jl

```diff
@@ -0,0 +1,67 @@
+struct WindowViewer{T<:Real, V<:AbstractVector{<:T}}
+    timeseries::V
+    halfwidth::Int
+    stride::Int
+    bracketing::Function
```

Comment:
Alright, let's discuss windowing a bit... First, windowing should be implemented with Dedicated Structs, like `Left, Right, Center` that are **singletons** (no fields). Then, you would implement the functions `bracketing_indices(i::Int, wv::WindowViewer{Center}` and same for `Left, Right`. The user definitely shouldn't have to give such low level functions.

But, the most important question to address first is: **Why do we even need different bracketing intervals / windows**? They are all equivalent, aren't they? Doesn't matter if the window is "left" or "right", it only leads to a translation of the rolling std's "time vector" by some indices. So, what's the gain? We could have only one window, the left one, and once we calculate `std_trend, std_tvec`, we can make the window magically be the right window by simply adjusting `std_tvec += window_size`.

Reply:
I agree, I'll leave out the bracketing for now then. The shifting operation you mention in case we want to have e.g. center bracketing can be implemented later if necessary.

Reply:
I also left out the `Left, Center, Right` singletons for now. Can still be introduced later.